code/handlers/finspaceservers.q

```diff
@@ -1,22 +1,16 @@
 .servers.FINSPACEDISC:@[value; `.servers.FINSPACEDISC; 0b];
 .servers.FINSPACECLUSTERSFILE:@[value; `.servers.FINSPACECLUSTERSFILE; hsym `];
 
-.servers.readclustersfile:{[]
-    if[null .servers.FINSPACECLUSTERSFILE; {.lg.e[`readclustersfile; "no finspace clusters file defined"]}];
-    :("SSSS**"; enlist ",") 0: .servers.FINSPACECLUSTERSFILE;
+.servers.listfinspaceclusters:{
+    :@[.aws.list_kx_clusters; `; {.lg.e[`listfinspaceclusters; "Failed to get finspace clusters using the finspace API - ",x]}];
     };
 
-.servers.getfinspaceclusters:{[]
-    expclusters:@[.servers.readclustersfile; `; {.lg.e[`getfinspaceclusters; "Failed to get read clusters.csv when using -  ",x]}];
-    availclusters:@[.aws.list_kx_clusters; `; {.lg.e[`getfinspaceclusters; "Failed to get finspace clusters using the finspace API - ",x]}];
-    :expclusters ij `cluster_name`cluster_type xkey select `$cluster_name, `$cluster_type, `$status from availclusters where status like "RUNNING";
-    };
-
-.servers.getfinspaceconn:{[ptype; pname]
-    id:.j.j[(ptype;pname)];
-    cluster:first exec cluster_name from .servers.getfinspaceclusters[] where proctype=ptype, procname = pname;
+.servers.getfinspaceconn:{[pname]
+    id:.Q.s1 pname;
+    runningclusters:select `$cluster_name, `$cluster_type, status from .servers.listfinspaceclusters[] where status like "RUNNING";
+    cluster:first exec cluster_name from runningclusters where pname = cluster_name;
```

Comment:
I think this can be simplified?

```suggestion
    cluster:first exec `$cluster_name from .servers.listfinspaceclusters[] where status like "RUNNING",cluster_name=pname;
```

(also minor thing but I think it's more conventional to put the column name on the left of a where clause comparison and the local variable being compared to on the right)